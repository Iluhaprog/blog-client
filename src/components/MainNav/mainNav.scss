@import '../../vars.scss';

.main-nav {
    padding: 20px 15px;
    box-shadow: $shadow-2;
    font-size: 14px;
    background-color: #fff;

    .user-short-info {
        margin: 0;
        padding: 0;

        .row {
            flex-direction: row-reverse;
        }

        p {
            color: #000;
            margin: 0;
            margin-right: 15px;
        }
        .round-box {
            width: 35px;
            height: 35px;
        }
    }

    nav {
        ul {
            list-style: none;
            padding: 0;
            margin: 0;
            li {
                margin: 0 15px;  

                a {
                    font-size: 16px;
                    text-decoration: none;
                    color: $text-color-6;
                    font-weight: 300;

                    &.selected {
                        color: $nav-bg;
                        text-decoration: underline;
                    }
                }
            }
        }
    }
}