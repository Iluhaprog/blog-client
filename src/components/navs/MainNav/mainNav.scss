@import '../../../vars.scss';

.main-nav {
    padding: 20px 15px;
    box-shadow: $shadow-2;
    font-size: 14px;
    background-color: #fff;

    .burger-button {
        display: none,
    }

    .user-short-info {
        margin: 0;
        padding: 0;

        .row {
            flex-direction: row-reverse;
        }

        p {
            color: #000;
            margin: 0;
            margin-right: 15px;
        }
        .round-box {
            width: 35px;
            height: 35px;
        }
    }

    nav {
        ul {
            list-style: none;
            padding: 0;
            margin: 0;
            li {
                margin: 0 15px;  

                a {
                    font-size: 16px;
                    text-decoration: none;
                    color: $text-color-6;
                    font-weight: 300;

                    &.selected {
                        color: $nav-bg;
                        text-decoration: underline;
                    }
                }
            }
        }
    }

    @media (max-width: 412px) {
        position: relative;
        padding: 5px;

        .burger-button {
            display: block;
            position: relative;
            width: 30px;
            height: 25px;

            span {
                position: absolute;
                top: 0;
                display: block;
                width: 100%;
                border: $border-1;
                transition: .3s ease-in-out;

                &:nth-child(2) {
                    top: 50%;
                }

                &:last-child {
                    top: 100%;
                }
            }

            &_visible {
                span:first-child {
                    top: 50%;
                    transform: rotate(45deg);
                }
                span:nth-child(2) {
                    opacity: 0;
                }
                span:last-child { 
                    top: 50%;
                    transform: rotate(-45deg);
                }
            }
        }

        nav {
            position: absolute;
            width: 100vw;
            height: calc(100vh - 45px);
            top: 45px;
            left: -100vw;
            background-color: rgba(255, 255, 255, .3);
            transition: .3s ease-in-out;
            z-index: 1;

            &.visible {
                left: 0;
            }
            
            ul {
                margin: 0 auto;
                max-width: 150px;
                flex-wrap: wrap;
                padding-top: 100px;

                li {
                    width: 100%;
                    margin: 20px;
                    text-align: center;
                }
            }
        }
    }
}