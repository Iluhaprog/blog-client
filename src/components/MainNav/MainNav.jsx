import React from 'react';
import { NavLink, useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import { PrimaryButton } from '../buttons';
import { Row } from '../containers';
import { UserShortInfo } from '../UserShortInfo';
import './mainNav.scss';

function MainNav(props) {
    const history = useHistory();
    const links = [
        {path: '/home', name: 'Home' },
        {path: '/search', name: 'Search'},
        {path: '/about', name: 'About'},
    ];
    return (
        <div className='main-nav'>
            <Row justifyContent='sb' alignItems='c'>
                <nav>
                    <ul className='row'>
                        {
                            links.map(link => (
                                <li key={link.path}>
                                    <NavLink to={link.path} activeClassName='selected'>
                                        {link.name}
                                    </NavLink>
                                </li>
                            ))
                        }    
                    </ul>                    
                </nav>
                {
                    props.isAuth 
                        ? (
                            <NavLink to='/profile'>
                                <UserShortInfo 
                                    row={true}  
                                    width={20}
                                    height={20}
                                />
                            </NavLink>
                        )
                        : (
                            <PrimaryButton 
                                text='Login'
                                onClick={() => {
                                    history.push('/login');
                                }}
                            />
                        )
                }
            </Row>
        </div>
    );
}

MainNav.defaultProps = {
    isAuth: false,
};

MainNav.propTypes = {
    isAuth: PropTypes.bool,
};

export default MainNav;