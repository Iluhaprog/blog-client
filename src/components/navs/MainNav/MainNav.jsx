import React, {useState} from 'react';
import { NavLink, useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import { PrimaryButton } from '../../buttons';
import { Row } from '../../containers';
import { UserShortInfo } from '../../UserShortInfo';
import './mainNav.scss';

function MainNav(props) {
    const history = useHistory();
    const [menuIsVisible, setMenuVisible] = useState(false);
    const links = [
        {path: '/home', name: 'Home' },
        {path: '/search', name: 'Search'},
        {path: '/about', name: 'About'},
    ];
    return (
        <div className='main-nav'>
            <Row justifyContent='sb' alignItems='c'>
                <div 
                    className={`burger-button ${menuIsVisible ? 'burger-button_visible' : ''}`}
                    onClick={() => setMenuVisible(!menuIsVisible)}
                >
                    <span></span>
                    <span></span>
                    <span></span>
                </div>
                <nav className={menuIsVisible ? 'visible' : ''}>
                    <ul className='row'>
                        {
                            links.map(link => (
                                <li key={link.path}>
                                    <NavLink 
                                        to={link.path} 
                                        activeClassName='selected'
                                        onClick={() => setMenuVisible(!menuIsVisible)}
                                    >
                                        {link.name}
                                    </NavLink>
                                </li>
                            ))
                        }    
                    </ul>                    
                </nav>
                {
                    props.isAuth 
                        ? (
                            <NavLink to='/profile'>
                                <UserShortInfo 
                                    row={true}  
                                    width={20}
                                    height={20}
                                />
                            </NavLink>
                        )
                        : (
                            <PrimaryButton 
                                text='Login'
                                onClick={() => {
                                    history.push('/login');
                                }}
                            />
                        )
                }
            </Row>
        </div>
    );
}

MainNav.defaultProps = {
    isAuth: false,
};

MainNav.propTypes = {
    isAuth: PropTypes.bool,
};

export default MainNav;