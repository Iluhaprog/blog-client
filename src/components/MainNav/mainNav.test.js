import React from 'react';
import configureMockStore from 'redux-mock-store'
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import MainNav from './MainNav';

const mockStore = configureMockStore();

test('Should return view (isAuth=false)', () => {
    const store = mockStore();
    const { asFragment } = render(
        <Provider store={store}>
            <BrowserRouter>
                <MainNav />
            </BrowserRouter>
        </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});

test('Should return view (isAuth=true)', () => {
    const store = mockStore({
        user: {
            firstName: 'Test firstname',
            lastName: 'Test lastname',
            avatarImage: '',
        },
    });
    const { asFragment } = render(
        <Provider store={store}>
            <BrowserRouter>
                <MainNav isAuth={true} />
            </BrowserRouter>
        </Provider>
    );
    expect(asFragment()).toMatchSnapshot(); 
});