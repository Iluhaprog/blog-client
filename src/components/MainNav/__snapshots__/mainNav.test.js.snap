// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should return view (isAuth=false) 1`] = `
<DocumentFragment>
  <div
    class="main-nav"
  >
    <div
      class="row row_jc-sb row_ai-c "
    >
      <nav>
        <ul
          class="row"
        >
          <li>
            <a
              href="/home"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/search"
            >
              Search
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
        </ul>
      </nav>
      <button
        class="button default-button primary-button"
      >
        Login
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Should return view (isAuth=true) 1`] = `
<DocumentFragment>
  <div
    class="main-nav"
  >
    <div
      class="row row_jc-sb row_ai-c "
    >
      <nav>
        <ul
          class="row"
        >
          <li>
            <a
              href="/home"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/search"
            >
              Search
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
        </ul>
      </nav>
      <a
        href="/profile"
      >
        <div
          class="user-short-info"
        >
          <div
            class="row  row_ai-c "
          >
            <div
              class="round-box"
            >
              <img
                alt="Test firstname-Test lastname"
                height="20"
                src="default.png"
                width="20"
              />
            </div>
            <p>
              Test firstname Test lastname
            </p>
          </div>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
